packages/mui-material/src/TablePagination/TablePagination.js

```diff
@@ -157,21 +158,24 @@ const TablePagination = React.forwardRef(function TablePagination(inProps, ref)
     SelectProps = {},
     showFirstButton = false,
     showLastButton = false,
+    slotProps,
     ...other
   } = props;
 
   const ownerState = props;
   const classes = useUtilityClasses(ownerState);
 
-  const MenuItemComponent = SelectProps.native ? 'option' : TablePaginationMenuItem;
+  const selectProps = slotProps?.select ?? SelectProps;
```

Comment:
Do we have tests for `SelectProps` to ensure we won't break the experience for other developers with this change?

Reply:
We have some tests [here](https://github.com/mui/material-ui/blob/edcb46b8e685282a1730f21bf8440687f6adde58/packages/mui-material/src/TablePagination/TablePagination.test.js#L382-L436). Do you think these would suffice, or should we add more?

Reply:
Those should be enough 👍🏼